addons/pos_paytm/models/pos_payment_method.py

```diff
@@ -0,0 +1,108 @@
+                    'issuerMaskCardNo': res_json['body'].get('issuerMaskCardNo'),
+                    'issuingBankName': res_json['body'].get('issuingBankName'),
+                    'payMethod': res_json['body'].get('payMethod'),
+                    'cardType': res_json['body'].get('cardType'),
+                    'cardScheme': res_json['body'].get('cardScheme'),
+                    'merchantReferenceNo': res_json['body'].get('merchantReferenceNo'),
+                    'merchantTransactionId': res_json['body'].get('merchantTransactionId'),
+                    'transactionDateTime': res_json['body'].get('transactionDateTime'),
+            })
+        return data
+
+    def paytm_make_payment_request(self, amount, transaction_id, reference_id, timestamp):
+        body = self._paytm_get_request_body(transaction_id, reference_id, timestamp)
+        body['transactionAmount'] = str(int(amount))
+        if self.accept_payment == 'auto':
+            body['autoAccept'] = 'True'
+        body['paymentMode'] = self.allowed_payment_modes.upper()
+        head = self._paytm_get_request_head(body)
+        merchantExtendedInfo = {'paymentMode': self.allowed_payment_modes.upper()}
+        if self.accept_payment == 'auto':
+            merchantExtendedInfo['autoAccept'] = 'True'
+        body['merchantExtendedInfo'] = merchantExtendedInfo
+        payload = {'head': head, 'body': body}
+        response = self._paytm_make_request('payment/request', payload=payload)
+        return response
+
+    def paytm_fetch_payment_status(self, transaction_id, reference_id, timestamp):
+        body = self._paytm_get_request_body(transaction_id, reference_id, timestamp)
+        head = self._paytm_get_request_head(body)
+        payload = {'head': head, 'body': body}
+        response = self._paytm_make_request('V2/payment/status', payload=payload)
+        return response
+
+    def _paytm_get_request_body(self, transaction_id, reference_id, timestamp):
+
+        time = datetime.fromtimestamp(timestamp).astimezone(tz=tz.gettz('Asia/Kolkata')).strftime("%Y-%m-%d %H:%M:%S")
+        return {
+            'paytmMid': self.paytm_mid,
+            'paytmTid': self.paytm_tid,
+            'transactionDateTime': time,
+            'merchantTransactionId': transaction_id,
+            'merchantReferenceNo': reference_id,
+        }
+
+    def _paytm_get_request_head(self, body):
+        return {
+            'requestTimeStamp' : body["transactionDateTime"],
+            'channelId' : self.channel_id,
+            'checksum' : paytm_utils.generate_signature(body, self.paytm_merchant_key),
+        }
+
+    @api.constrains('use_payment_terminal')
+    def _check_paytm_terminal(self):
+        for record in self:
+            if record.use_payment_terminal == 'paytm' and record.company_id.currency_id.name != 'INR':
+                raise UserError(_('This Payment Terminal is only valid for INR Currency'))
```

Comment:
you raise an error but the cashier has absolutely no idea what's going on.
The error must go up in the JS

Reply:
@qle-odoo it's not clear!!
this is Python constraints and it raises an error when the user is configuring the POS terminal not when using.

Reply:
yes my bad